src/main/java/EventsAndCommands/CommandLoader.java

```diff
@@ -0,0 +1,222 @@
+package EventsAndCommands;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.jagrosh.jdautilities.command.Command;
+
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+
+import org.reflections.*;
+import com.mongodb.client.MongoDatabase;
+import com.jagrosh.jdautilities.commons.waiter.EventWaiter;
+
+/**
+ * CommandHandler
+ */
+public class CommandLoader {
+
+    private List<Command> loadedCommands;
+    private List<ListenerAdapter> loadedListeners;
+    private MongoDatabase db;
+    private Properties config;
+    private EventWaiter waiter;
+
+    public CommandLoader(MongoDatabase db, Properties config, EventWaiter waiter) {
+        loadedCommands = new ArrayList<>();
+        loadedListeners = new ArrayList<>();
+        setDb(db);
+        setConfig(config);
+        setWaiter(waiter);
+    }
+
+    public void loadCommands() {
+        Reflections reflections = new Reflections(this.getClass().getPackageName());
+
+        Set<Class<? extends Command>> cmds = reflections.getSubTypesOf(Command.class);
+        cmds = cmds.stream()
+            .filter(x -> x.getAnnotation(IgnoreCommand.class) == null)
+            .collect(Collectors.toSet());
+
+        List<Command> instances = new ArrayList<>();
+
+        for (var cmd : cmds) {
+            List<Constructor<?>> ctors = Arrays.asList(cmd.getConstructors());
+            for (var ctor : ctors) {
+                if (hasAcceptableParams(ctor)) {
+                    try {
+                        instances.add(injectCommand(ctor));
+                    } catch (InvocationTargetException e){
+                        e.getCause().printStackTrace();
+                    } catch (Exception e) {
+                        // TODO: Log
+                        System.err.println(e.getMessage());
+                        System.err.println("Could not load " + cmd.getName());
+                    }
+                } else {
+                    System.err.println("No suitable constructor for " + cmd.getName());
+                }
+            }
+        }
+
+        setLoadedCommands(instances);
+    }
+
+    public void loadListenerAdaptor() {
```

Comment:
You can abstract the reflections+subtypes+filtering+collecting out in its own method.

Likely you can even extend that to deduplicate the two load methods into one.
*Might* be useful, even though you just use that pattern twice.